Installation via cargo blocked with `error[E0432]: unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6``

https://github.com/AustinWise/smeagol/blob/5259940ee00227861661a9b8513d4fceeb2cd5bd/README.md?plain=1#L26-L28

```console
> rustc --version; cargo --version
rustc 1.77.2 (25ef9e3d8 2024-04-09) (built from a source tarball)
cargo 1.77.1

> cargo install smeagol-wiki
    Updating crates.io index
  Installing smeagol-wiki v0.4.9
    Updating crates.io index
   Compiling proc-macro2 v1.0.89
   Compiling unicode-ident v1.0.13
   Compiling libc v0.2.161
   Compiling version_check v0.9.5
   Compiling serde v1.0.214
   Compiling shlex v1.3.0
   Compiling cfg-if v1.0.0
   Compiling memchr v2.7.4
   Compiling autocfg v1.4.0
   Compiling itoa v1.0.11
   Compiling pkg-config v0.3.31
   Compiling byteorder v1.5.0
   Compiling num-conv v0.1.0
   Compiling time-core v0.1.2
   Compiling once_cell v1.20.2
   Compiling pin-project-lite v0.2.15
   Compiling yansi v1.0.1
   Compiling bytes v1.8.0
   Compiling httparse v1.9.5
   Compiling futures-core v0.3.31
   Compiling fnv v1.0.7
   Compiling powerfmt v0.2.0
   Compiling time-macros v0.2.18
   Compiling hashbrown v0.15.0
   Compiling equivalent v1.0.1
   Compiling futures-sink v0.3.31
   Compiling proc-macro2-diagnostics v0.10.1
   Compiling cookie v0.18.1
   Compiling crossbeam-utils v0.8.20
   Compiling log v0.4.22
   Compiling slab v0.4.9
   Compiling ref-cast v1.0.23
   Compiling futures-channel v0.3.31
   Compiling percent-encoding v2.3.1
   Compiling futures-task v0.3.31
   Compiling futures-io v0.3.31
   Compiling bitflags v2.6.0
   Compiling pin-utils v0.1.0
   Compiling aho-corasick v1.1.3
   Compiling quote v1.0.37
   Compiling either v1.13.0
   Compiling regex-syntax v0.8.5
   Compiling futures-util v0.3.31
   Compiling jobserver v0.1.32
   Compiling syn v2.0.85
   Compiling signal-hook-registry v1.4.2
   Compiling mio v1.0.2
   Compiling socket2 v0.5.7
   Compiling is-terminal v0.4.13
   Compiling getrandom v0.2.15
   Compiling cc v1.1.31
   Compiling zstd-safe v6.0.6
   Compiling stable_deref_trait v1.2.0
   Compiling vcpkg v0.2.15
   Compiling rustix v0.38.38
   Compiling crunchy v0.2.2
   Compiling ownedbytes v0.5.0
   Compiling rand_core v0.6.4
   Compiling deranged v0.3.11
   Compiling http v0.2.12
   Compiling tracing-core v0.1.32
   Compiling uncased v0.9.10
   Compiling ahash v0.8.11
   Compiling unicase v2.8.0
   Compiling try-lock v0.2.5
   Compiling tinyvec_macros v0.1.1
   Compiling linux-raw-sys v0.4.14
   Compiling smallvec v1.13.2
   Compiling time v0.3.36
   Compiling mime_guess v2.0.5
   Compiling tinyvec v1.8.0
   Compiling http-body v0.4.6
   Compiling want v0.3.1
   Compiling tracing v0.1.40
   Compiling regex-automata v0.4.8
   Compiling tokio v1.41.0
   Compiling crossbeam-epoch v0.9.18
   Compiling zstd-sys v2.0.13+zstd.1.5.6
   Compiling libz-sys v1.1.20
   Compiling ring v0.16.20
   Compiling lock_api v0.4.12
   Compiling inlinable_string v0.1.15
   Compiling libm v0.2.11
   Compiling minimal-lexical v0.2.1
   Compiling utf8-ranges v1.0.5
   Compiling utf8parse v0.2.2
   Compiling tower-service v0.3.3
   Compiling parking_lot_core v0.9.10
   Compiling winnow v0.6.20
   Compiling regex-syntax v0.6.29
   Compiling rayon-core v1.12.1
   Compiling httpdate v1.0.3
   Compiling futures v0.3.31
   Compiling anstyle-parse v0.2.6
   Compiling nom v7.1.3
   Compiling indexmap v2.6.0
   Compiling crossbeam-deque v0.8.5
   Compiling unicode-normalization v0.1.24
   Compiling devise_core v0.4.2
   Compiling libgit2-sys v0.16.2+1.7.2
   Compiling bitpacking v0.8.4
   Compiling regex v1.11.1
   Compiling hyper v0.14.31
   Compiling tantivy-fst v0.4.0
   Compiling stable-pattern v0.1.0
   Compiling num-traits v0.2.19
   Compiling figment v0.10.19
   Compiling multer v3.1.0
   Compiling thiserror v1.0.65
   Compiling scopeguard v1.2.0
   Compiling mime v0.3.17
   Compiling untrusted v0.7.1
   Compiling anstyle-query v1.1.2
   Compiling colorchoice v1.0.3
   Compiling anstyle v1.0.9
   Compiling spin v0.5.2
   Compiling fastrand v2.1.1
   Compiling unicode-bidi v0.3.17
   Compiling murmurhash32 v0.3.1
   Compiling is_terminal_polyfill v1.70.1
   Compiling serde_json v1.0.132
   Compiling rustversion v1.0.18
   Compiling state v0.6.0
   Compiling anstream v0.6.17
   Compiling tempfile v3.13.0
   Compiling serde_derive v1.0.214
   Compiling zerocopy-derive v0.7.35
   Compiling tokio-macros v2.4.0
   Compiling pear_codegen v0.2.9
   Compiling async-trait v0.1.83
   Compiling ref-cast-impl v1.0.23
   Compiling devise_codegen v0.4.2
   Compiling async-stream-impl v0.3.6
   Compiling pear v0.2.9
   Compiling zerocopy v0.7.35
   Compiling thiserror-impl v1.0.65
   Compiling devise v0.4.2
   Compiling idna v0.5.0
   Compiling rocket_http v0.5.1
   Compiling ppv-lite86 v0.2.20
   Compiling askama_parser v0.2.1
   Compiling hashbrown v0.13.2
   Compiling rand_chacha v0.3.1
   Compiling rand v0.8.5
   Compiling itertools v0.10.5
   Compiling num_cpus v1.16.0
   Compiling form_urlencoded v1.2.1
   Compiling http v1.1.0
   Compiling combine v4.6.7
   Compiling rocket v0.5.1
   Compiling encoding_rs v0.8.35
   Compiling instant v0.1.13
   Compiling fastdivide v0.4.1
   Compiling glob v0.3.1
   Compiling spin v0.9.8
   Compiling pulldown-cmark v0.9.6
   Compiling termcolor v1.4.1
   Compiling clap_lex v0.7.2
   Compiling strsim v0.11.1
   Compiling unicode-xid v0.2.6
   Compiling ryu v1.0.18
   Compiling winnow v0.5.40
   Compiling heck v0.5.0
   Compiling humantime v2.1.0
   Compiling clap_derive v4.5.18
   Compiling clap_builder v4.5.20
   Compiling rocket_codegen v0.5.1
   Compiling env_logger v0.10.2
   Compiling measure_time v0.8.3
   Compiling fail v0.5.1
   Compiling tokio-util v0.7.12
   Compiling tokio-stream v0.1.16
   Compiling url v2.5.2
   Compiling toml_datetime v0.6.8
   Compiling serde_spanned v0.6.8
   Compiling basic-toml v0.1.9
   Compiling sketches-ddsketch v0.2.2
   Compiling tantivy-bitpacker v0.4.0
   Compiling tantivy-tokenizer-api v0.1.1
   Compiling ubyte v0.10.4
   Compiling rust-stemmers v1.2.0
   Compiling h2 v0.3.26
   Compiling toml_edit v0.22.22
   Compiling askama_derive v0.12.5
   Compiling toml_edit v0.19.15
   Compiling tantivy-query-grammar v0.20.0
   Compiling uuid v1.11.0
   Compiling humansize v2.1.3
   Compiling parking_lot v0.12.3
   Compiling lru v0.10.1
   Compiling tantivy-common v0.5.0
   Compiling tantivy-stacker v0.1.0
   Compiling toml v0.8.19
   Compiling rayon v1.10.0
   Compiling smeagol-wiki v0.4.9
   Compiling async-stream v0.3.6
   Compiling fs4 v0.6.6
   Compiling crossbeam-channel v0.5.13
   Compiling memmap2 v0.6.2
   Compiling crc32fast v1.4.2
   Compiling oneshot v0.1.8
   Compiling lz4_flex v0.10.0
   Compiling arc-swap v1.7.1
   Compiling htmlescape v0.3.1
   Compiling base64 v0.21.7
   Compiling rustc-hash v1.1.0
   Compiling downcast-rs v1.2.1
   Compiling levenshtein_automata v0.2.1
   Compiling census v0.4.2
   Compiling askama_escape v0.10.3
   Compiling atomic v0.5.3
   Compiling binascii v0.1.4
   Compiling heck v0.4.1
   Compiling askama v0.12.1
   Compiling strum_macros v0.25.3
   Compiling toml v0.7.8
   Compiling clap v4.5.20
   Compiling pretty_env_logger v0.5.0
   Compiling bitflags v1.3.2
   Compiling strum v0.25.0
   Compiling lazy_static v1.5.0
   Compiling git2 v0.18.3
error[E0432]: unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6`
   --> /home/kachick/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zstd-safe-6.0.6/src/lib.rs:609:13
    |
609 |             ZSTD_c_experimentalParam6 as ZSTD_c_targetCBlockSize,
    |             -------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             no `ZSTD_c_experimentalParam6` in `ZSTD_cParameter`
    |             help: a similar name exists in the module: `ZSTD_c_experimentalParam1`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `zstd-safe` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `smeagol-wiki v0.4.9`, intermediate artifacts can be found at `/tmp/cargo-installuzULhy`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,6 +32,6 @@ rocket = "0.5.0"
 serde = { version = "1.0", features = ["derive"] }
 strum = "0.25"
 strum_macros = "0.25"
-tantivy = "0.20"
+tantivy = "0.22"
 thiserror = "1.0"
 toml = "0.7"

--- a/src/wiki.rs
+++ b/src/wiki.rs
@@ -307,7 +307,7 @@ impl Wiki {
         Ok(top_docs
             .iter()
             .filter_map(|(score, doc_address)| {
-                let doc = match searcher.doc(*doc_address) {
+                let doc = match searcher.doc::<TantivyDocument>(*doc_address) {
                     Ok(doc) => doc,
                     Err(_) => return None,
                 };
@@ -317,13 +317,13 @@ impl Wiki {
                 let title = doc
                     .get_first(fields.title)
                     .unwrap()
-                    .as_text()
+                    .as_str()
                     .unwrap()
                     .to_owned();
                 let path = doc
                     .get_first(fields.path)
                     .unwrap()
-                    .as_text()
+                    .as_str()
                     .unwrap()
                     .to_owned();
                 let snippet_html = highlight(snippet);

--- a/src/wiki.rs
+++ b/src/wiki.rs
@@ -286,7 +286,7 @@ impl Wiki {
             .0
             .index
             .reader_builder()
-            .reload_policy(ReloadPolicy::OnCommit)
+            .reload_policy(ReloadPolicy::OnCommitWithDelay)
             .try_into()?;
 
         let searcher = reader.searcher();

--- a/src/wiki.rs
+++ b/src/wiki.rs
@@ -141,7 +141,7 @@ fn index_file(
         url += "/";
         url += path;
     }
-    let mut doc = Document::default();
+    let mut doc = TantivyDocument::default();
     doc.add_text(search_fields.path, &url);
     doc.add_text(search_fields.title, page.title);
     doc.add_text(search_fields.body, page.body);

